### PR TITLE
MGMT-17295: Deleting a fleet without selectors crashes the server

### PR DESCRIPTION
**Pull request by avishayt**

### CODE REVIEW

File: internal/tasks/fleet_selector_test.go [hunk 1]
```diff
@@ -106,14 +106,29 @@ var _ = Describe("FleetSelector", func() {
 			Expect(device.Metadata.Owner).To(BeNil())
 		})
 
-		It("Fleet selector removed no overlap", func() {
+		It("Nil fleet selector matches all", func() {
```
Comment (mangelajo) [inline internal/tasks/fleet_selector_test.go:109]: We should make those 
```suggestion
		It("should match all for nil fleet selector", func() {
```

File: internal/tasks/fleet_selector_test.go [hunk 2]
```diff
@@ -106,14 +106,29 @@ var _ = Describe("FleetSelector", func() {
...
+			listParams := store.ListParams{Limit: 0}
+			devices, err := deviceStore.List(ctx, orgId, listParams)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(devices.Items)).To(Equal(4))
+			for _, device := range devices.Items {
+				Expect(*device.Metadata.Owner).To(Equal("Fleet/fleet"))
+			}
+
+			callback := store.FleetStoreCallback(func(before *model.Fleet, after *model.Fleet) {})
+			err = fleetStore.Delete(ctx, orgId, callback, "fleet")
```
Comment (mangelajo) [inline internal/tasks/fleet_selector_test.go:127]:
```suggestion
			err = fleetStore.Delete(ctx, orgId, callback, "fleet")
```

I would suggest we move deletion to a separate test

It("should delete a fleet will nil selectors without errors")

so if something goes wrong with deletion, we have  more granular test output

File: internal/tasks/fleet_selector_test.go (hunk 1 above)
Comment (avishayt) [reply internal/tasks/fleet_selector_test.go:109]: Why?  This description is more difficult for me to understand.
How about "Nil fleet selector should match all devices"?

File: internal/tasks/fleet_selector_test.go (hunk 2 above)
Comment (avishayt) [reply internal/tasks/fleet_selector_test.go:127]: Done

File: internal/tasks/fleet_selector_test.go (hunk 1 above)
Comment (avishayt) [reply internal/tasks/fleet_selector_test.go]: Nice reference, now I understand what you meant.  I'll do an overhaul in a separate PR.

**Comment by mangelajo** [comment]
As we commented on slack, let's merge, and then we can do a more ginkgo/idiomatic overhaul

**Pull request merged**